packages/rspack/src/trace/index.ts

```diff
@@ -92,20 +92,17 @@ export class JavaScriptTracer {
 					}
 				});
 			}
-			const originTrace = fs.readFileSync(this.output, "utf-8");
 			// this is hack, [] is empty and [{}] is not empty
-			const originTraceIsEmpty = !originTrace.includes("{");
-			const eventMsg =
-				(this.events.length > 0 && !originTraceIsEmpty ? "," : "") +
-				this.events
-					.map(x => {
-						return JSON.stringify(x);
-					})
-					.join(",\n");
+			const eventMsg = this.events
```

Comment:
if eventMsg's length hit the limit, it's better to write the content by fs stream api .

Reply:
stream is async I change it to manually stream write to avoid intermediate large string